runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,49 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview` / `start`, or `open` commands
+--- depending on the system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---
+---@return string returns the string output of the command. See |system()|.
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri)
+  local cmd
+
+  if vim.fn.has("wsl") then
+    cmd = "wslview"
+  elseif vim.fn.has("mac") then
+    cmd = 'open'
+  elseif vim.fn.has("win32") then
+    cmd = "start"
+  else -- Assume linux
+    cmd = 'xdg-open'
+  end
+
+  local ret = vim.fn.system({
```

Comment:
Better to use `jobstart` here. `vim.fn.system` sometimes fails for unknown reasons. See here for example https://github.com/folke/lazy.nvim/issues/18

```lua
  local ret = vim.fn.jobstart(cmd, { detach = true })
  if ret <= 0 then
    local msg = {
      "Failed to open uri",
      ret,
      vim.inspect(cmd),
    }
    vim.notify(table.concat(msg, "\n"), vim.log.levels.ERROR)
  end
```

Reply:
I accidentally requestes changes, but these are obviously more suggestions